src/rust/en.mangairo/src/parser.rs

```diff
@@ -0,0 +1,376 @@
+
+		let id = obj.select(".story-name a").attr("href").read();
+		let title = obj.select(".story-name a ").text().read();
+		let img = obj.select(".story-list-img img").attr("src").read();
+
+		if !id.is_empty() && !title.is_empty() && !img.is_empty() {
+			result.push(Manga {
+				id,
+				cover: img,
+				title,
+				..Default::default()
+			});
+		}
+	}
+
+	// Example: 'Total: 38,202 stories'
+	let mut total_str = html.select(".quantitychapter").text().read();
+	total_str = total_str.replace("Total: ", "");
+	total_str = total_str.replace(" stories", "");
+	total_str = total_str.chars().filter(|&c| c != ',').collect();
+
+	let total_result = total_str.parse::<i32>();
+	if let Ok(total) = total_result {
+		Some(total)
+	} else {
+		None
+	}
+}
+
+pub fn parse_manga_details(html: Node, id: String) -> Result<Manga> {
+	let title = html
+		.select(".breadcrumbs p span a span")
+		.last()
+		.text()
+		.read();
+	let cover = html.select(".avatar").attr("src").read();
+	let description = html
+		.select("div#story_discription p")
+		.text()
+		.read()
+		.trim()
+		.to_string();
+	let status_str = html
+		.select(".story_info_right li:nth-child(5) a")
+		.text()
+		.read()
+		.to_lowercase();
+
+	let url = format!("{}", &id);
+
+	let mut authors: Vec<String> = Vec::new();
+	html.select(".story_info_right li:nth-child(3) a")
+		.array()
+		.for_each(|tag| {
+			authors.push(String::from(
+				tag.as_node().expect("node array").text().read().trim(),
+			))
+		});
+	let author = authors.join(", ");
+
+	let mut categories: Vec<String> = Vec::new();
+	html.select(".story_info_right .a-h")
+		.array()
+		.for_each(|tag| categories.push(tag.as_node().expect("node array").text().read()));
+
+	let status = match status_str.as_str() {
+		"ongoing" => MangaStatus::Ongoing,
+		"completed" => MangaStatus::Completed,
+		_ => MangaStatus::Unknown,
+	};
+
+	let nsfw = if categories.contains(&String::from("Pornographic"))
+		|| categories.contains(&String::from("Adult"))
+		|| categories.contains(&String::from("Smut"))
+		|| categories.contains(&String::from("Erotica"))
+	{
+		MangaContentRating::Nsfw
+	} else if categories.contains(&String::from("Ecchi")) {
+		MangaContentRating::Suggestive
+	} else {
+		MangaContentRating::Safe
+	};
+
+	let viewer = if categories.contains(&String::from("Manhua"))
+		|| categories.contains(&String::from("Manhwa"))
+		|| categories.contains(&String::from("Webtoons"))
+	{
+		MangaViewer::Scroll
+	} else {
+		MangaViewer::Rtl
+	};
+
+	Ok(Manga {
+		id,
+		cover,
+		title,
+		author,
+		description,
+		url,
+		categories,
+		status,
+		nsfw,
+		viewer,
+		..Default::default()
+	})
+}
+
+pub fn get_chapter_list(html: Node) -> Result<Vec<Chapter>> {
+	let mut chapters: Vec<Chapter> = Vec::new();
+	for chapter in html.select(".chapter_list ul li a").array() {
+		let obj = chapter.as_node().expect("node array");
+		let url = obj.attr("href").read();
+		let id = parse_incoming_url_chapter_id(url.clone());
+
+		if let Some(id_value) = id {
```

Comment:
You don't need to rename things when you unwrap them
```suggestion
		if let Some(id) = id {
```